tests/unit/amo/pages/TestUserProfileEdit.js

```diff
@@ -884,39 +886,54 @@ describe(__filename, () => {
     ).not.toBeDisabled();
   });
 
-  it('closes the modal when user clicks the cancel button', () => {
+  it('closes the modal when user clicks the cancel button', async () => {
     renderForCurrentUser();
 
     userEvent.click(screen.getByRole('button', { name: 'Delete My Profile' }));
 
-    const button = screen.getByRole('button', { name: 'Cancel' });
+    let button;
+    await waitFor(() => {
+      button = screen.getByRole('button', { name: 'Cancel' });
+      expect(button).toBeInTheDocument();
+    });
     const clickEvent = createEvent.click(button);
     const preventDefaultWatcher = jest.spyOn(clickEvent, 'preventDefault');
 
     fireEvent(button, clickEvent);
 
     expect(preventDefaultWatcher).toHaveBeenCalled();
-    expect(
-      screen.queryByText(
-        'IMPORTANT: Deleting your Firefox Add-ons profile is irreversible.',
-      ),
-    ).not.toBeInTheDocument();
+    await waitFor(() =>
+      expect(
+        screen.queryByText(
+          'IMPORTANT: Deleting your Firefox Add-ons profile is irreversible.',
+        ),
+      ).not.toBeInTheDocument(),
+    );
   });
 
-  it('dispatches deleteUserAccount and logOutUser when current logged-in user confirms account deletion', () => {
+  it('dispatches deleteUserAccount and logOutUser when current logged-in user confirms account deletion', async () => {
     const userId = signInUserWithProps(userProps);
-    const history = createHistory({
-      initialEntries: [getLocation(userId)],
-    });
-    const pushSpy = jest.spyOn(history, 'push');
     const dispatch = jest.spyOn(store, 'dispatch');
-    render({ history });
+
+    render({ initialEntries: [getLocation(userId)] });
```

Comment:
This is the problem. `render` does not accept an `initialEntries` argument, it accepts either `location` or `userId`. In this case, you can just pass `userId` and it will work (because that will end up generating `getLocation(userId)`).